src/RemoteRequest/TemporaryFileCachedRemoteGetRequest.php

```diff
@@ -0,0 +1,156 @@
+<?php
+
+namespace AmpProject\RemoteRequest;
+
+use AmpProject\Exception\FailedToGetCachedResponse;
+use AmpProject\RemoteGetRequest;
+use AmpProject\Response;
+use DateTimeImmutable;
+
+/**
+ * Temporarily cache remote response.
+ *
+ * @package ampproject/amp-toolbox
+ */
+final class TemporaryFileCachedRemoteGetRequest implements RemoteGetRequest
```

Comment:
The `TemporaryFileCachedRemoteGetRequest` class should be implemented as a [*Decorator*](https://refactoring.guru/design-patterns/decorator/php/example), which receives a `RemoteGetRequest` implementation and wraps the caching functionality around that.